lib/core/config/dependencies.dart

```diff
@@ -86,4 +87,31 @@ Future<void> get _fixtures async {
       repository: sl(),
     ),
   );
+}
+
+Future<void> get _prediction async {
+  sl.registerFactory(
+    () => PredictionBloc(
+      useCase: sl(),
+    ),
+  );
+
+  sl.registerLazySingleton<PredictionRepository>(
+    () => PredictionRepositoryImpl(
+      network: sl(),
+      remote: sl(),
+    ),
+  );
+
+  sl.registerLazySingleton<PredictionRemoteDataSource>(
+    () => PredictionDataSourceImpl(
+      client: sl(),
+    ),
+  );
+
+  sl.registerLazySingleton<PredictionUseCase>(
```

Comment:
UseCase should be `factory`

Use this snippet for the solution

```dart 
sl.registerFactory(
    () => PredictionUseCase(
      repository: sl(),
    ),
  );
```

Reply:
done